unitTests/testTensorOpsFixedSize.cpp

```diff
@@ -351,6 +351,113 @@ class FixedSizeSquareMatrixTest : public ::testing::Test
         } );
   }
 
+  void testSymRij_eq_AiAj()
+  {
+    T result[ N ][ N ];
+    for( std::ptrdiff_t i = 0; i < N; ++i )
+    {
+      for( std::ptrdiff_t j = 0; j < N; ++j )
+      {
+        result[ i ][ j ] = m_vectorA_local[ i ] * m_vectorA_local[ j ];
+      }
+    }
+
+    T symResult[ SYM_SIZE ];
+    tensorOps::denseToSymmetric< N >( symResult, result );
+
+    ArrayViewT< T, 2, 1 > const symMatrixA_IJ = m_symMatrixA_IJ.toView();
+    ArrayViewT< T, 2, 0 > const symMatrixA_JI = m_symMatrixA_JI.toView();
+
+    ArrayViewT< T const, 2, 1 > const vectorA_IJ = m_vectorA_IJ.toViewConst();
+    ArrayViewT< T const, 2, 0 > const vectorA_JI = m_vectorA_JI.toViewConst();
+    T const ( &vectorA_local )[ N ] = m_vectorA_local;
+
+    std::ptrdiff_t const matrixSeed = m_seedSymMatrixA;
+
+    forall< POLICY >( 1, [symResult, symMatrixA_IJ, symMatrixA_JI, vectorA_IJ, vectorA_JI, vectorA_local, matrixSeed]
+                      LVARRAY_HOST_DEVICE ( int )
+        {
+          #define _TEST( symMatrix, vectorN ) \
+            fill( symMatrix, matrixSeed ); \
+            tensorOps::symRij_eq_AiAj< N >( symMatrix, vectorN ); \
+            CHECK_EQUALITY_1D( SYM_SIZE, symMatrix, symResult )
+
```

Comment:
@corbett5 since that matrix gets overwritten I wonder if it is necessary here to use all the permutations of `symMatrix` and also if that `fill` is actually necessary.

Reply:
The call to `fill` is necessary because it `symMatrix` is re-used between calculations each of which should produce the same result. If there was no call to fill, the first calculation might work but then subsequent ones could do nothing and the tests would pass.

As for the permutations they're mostly there for compile time checks.